test/e2e/cmd/run/job_manager.go

```diff
@@ -128,24 +129,31 @@ func (jm *JobsManager) Start() {
 				log.Error(errors.New("received an update event for an unmanaged Pod"), "namespace", newPod.Namespace, "name", newPod.Name)
 				return
 			}
+
 			switch newPod.Status.Phase {
 			case corev1.PodRunning:
 				job.onPodEvent(jm.Clientset, newPod)
-			case corev1.PodSucceeded:
-				if job.podSucceeded {
-					// already done, but the informer is not stopped yet so this code is still running
-					return
+
+				// download result file when pod is ready
+				if podutils.IsPodReady(newPod) {
```

Comment:
Is that really worth the extra `kubectl` dependency? I don't see a big difference to our existing `k8s.IsPodReady` (we also check for `ContainersReady`)

https://github.com/elastic/cloud-on-k8s/blob/b6b50758127d24c1e67b62799711864fcd138582/pkg/utils/k8s/k8sutils.go#L67-L75

Reply:
Really not, I'll use ours.